src/game_inventory.cpp

```diff
@@ -548,10 +548,7 @@ class comestible_inventory_preset : public inventory_selector_preset
                 if( calories_per_effective_volume == 0 ) {
                     return std::string();
                 }
-                /* This is for screen readers. I will make a PR to discuss what these prerequisites could be -
-                bio_digestion, selfaware, high cooking skill etc*/
-                constexpr bool ARBITRARY_PREREQUISITES_TO_BE_DETERMINED_IN_THE_FUTURE = false;
-                if( ARBITRARY_PREREQUISITES_TO_BE_DETERMINED_IN_THE_FUTURE ) {
+                if( p.has_trait( trait_id( "SELFAWARE" ) ) || ( p.get_skill_level( skill_id( "COOKING" )) > 6 )  ||  p.has_bionic( bionic_id( "bio_digestion" ) ) ) {
                     return string_format( "%d", calories_per_effective_volume );
                 }
                 std::string result = satiety_bar( calories_per_effective_volume );
```

Comment:
I would stuff the *whole* 
```c++
                if( ...) {
                    return ...;
                }
```
block **into** `satiety_bar` method. That way you will _always_ get numbers if numbers should be used (and you don't have to special case each call site) and it also allows _easy_ colorization for the raw number (because I think that's a nifty feature that doesn't deserve to be thrown away just because the character is more knowledgeable about nutrition).

Reply:
Sounds like a  great idea, but the colorization of the integer value is not trivial, since it was previously based on an arbitrary upper cap.  Nevertheless, I'll try and do this as it helps with the development of #44565.

Reply:
> is not trivial, since it was previously based on an arbitrary upper cap.

Huh? `get_bar` returns a _pair_. Ignore the first "|||\" part, but use the second one: color.

```c++
...
auto bar = get_bar(...);
if (fancy_condition) {
  return colorize(string_format("%g", raw_number), bar.second);
}
return bar.first;
...
```

Reply:
Yeah you're probably right. I'll check it out once I'm done with some tweaks I'm doing over at #44565.

Reply:
This "simple" refactor turned out to be much more complicated than I thought. I needed to move functions around (from output.cpp/h to game_inventory.cpp/h) and change the argument list to include a reference to player_character in order to check player skills and bionics. Furthermore, this broke the nice suite of tests @wapcaplet had created for me, so I had to recreate a function overload that did NOT take player_character. All in all, it's a huge mess. Look at my commit descriptions for more info.